internal/controller/server_controller.go

```diff
@@ -516,10 +546,35 @@ func (r *ServerReconciler) applyDefaultIgnitionForServer(ctx context.Context, lo
 	return nil
 }
 
-func (r *ServerReconciler) generateDefaultIgnitionDataForServer(flags string) ([]byte, error) {
-	ignitionData, err := ignition.GenerateDefaultIgnitionData(ignition.ContainerConfig{
-		Image: r.ProbeImage,
-		Flags: flags,
+func generateSSHKeyPairs() ([]byte, []byte, error) {
+	privateKey, err := rsa.GenerateKey(rand.Reader, 2048)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to generate private key: %w", err)
+	}
+
+	privateKeyBytes := x509.MarshalPKCS1PrivateKey(privateKey)
+	privateKeyPem := pem.EncodeToMemory(&pem.Block{
+		Type:  "RSA PRIVATE KEY",
+		Bytes: privateKeyBytes,
+	})
+
+	publicKey := &privateKey.PublicKey
+	publicKeyBytes, err := x509.MarshalPKIXPublicKey(publicKey)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to marshal public key: %w", err)
+	}
+	publicKeyPem := pem.EncodeToMemory(&pem.Block{
+		Type:  "PUBLIC KEY",
+		Bytes: publicKeyBytes,
+	})
```

Comment:
Does `ssh.MarshalAuthorizedKey(publicKey)` achieve the same?

Reply:
Already fixed with the last commit.